[feat]: brand new enums (#69)
soramitsu/scale-codec-js-library: packages/enum/README.md

# `@scale-codec/enum` ![build status](https://img.shields.io/github/checks-status/soramitsu/scale-codec-js-library/master) ![version](https://img.shields.io/npm/v/@scale-codec/enum) ![license](https://img.shields.io/npm/l/@scale-codec/enum)

Small tagged-union library for TypeScript.

[//]: # 'TODO'
[//]: # '[Documentation](https://soramitsu.github.io/scale-codec-js-library/guide/enum)'

## Features

> - Follows TypeScript's [discriminated union pattern](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions), which leads to **type narrowing** and **exhaustiveness check** support.
> - Type-safe variants creation with type inference support

Reviewer: ```suggestion
- **Type narrowing** and **exhaustiveness check** (due to the library following TypeScript's [discriminated union pattern](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions))
- Type-safe variants creation with type inference
```
